Problem with @ng-toolkit/_utils

I really don't know what's the issue. Probably `@ng-toolkit/_utils` it's not published in the proper way. Unfortunately, all the packages which depend on those, are unable to properly consume this package, which leads to:

```
Error: Cannot find module '@ng-toolkit/_utils/testing
```

The problem doesn't occur on my local environment (by running ./scripts/deploy_verdaccio.sh), it occurs only on the Travis CI.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,8 @@
 language: node_js
 node_js: 10
 script:
-- if [ "$TRAVIS_BRANCH" = "master" ]; then ./scripts/deploy.sh; fi
-- if [ "$TRAVIS_BRANCH" != "master" ]; then ./scripts/deploy_verdaccio.sh; fi
+  - if [ "$TRAVIS_BRANCH" = "master" ]; then ./scripts/deploy.sh; fi
+  - if [ "$TRAVIS_BRANCH" != "master" ]; then ./scripts/deploy_verdaccio.sh; fi
 env:
   global:
     secure: yPhiY6aqwJ0cDLVYLvY6g9r8T1W/mVh7oFXnQS+Qqp5YUTDGI2O6SKGMzQeMpV8mtql5vyeGgWtEyZ5aZqsX83LJFpSY5375ClcwIA2MIMCzeici3jqt00BnXW1t75U1i4F5T8THbKKc4IzvS3ORTKwJo+lZx4dZR5JBagI1sZAU++mO+aXEOJUwfOCbE/cDHh4ROC8e0D4UOfwp5Bk9NQ4K6P5Rl/aIwSc/77AVh95qzlXkg+V6py9jJgg1F6dl62XHLft8uNC/67oxHXIOWTcu4s11ob0cBza33DzwSa1nYX1jRdMd9Uq0gU8MxGulkv6RXwju7wQw4btY6eKqD+da1BisA14kUOaETPy04chVlyQGMUDzs77mYH8/A6TiNDBnZM+nlz6gnwutfMJKr6mRq1Wwsk5XijckFPmgoHj7B+vTz6gN6KtJ46PTR58LLEqWpILpLwOD430Qce6zfjbuAZKJiTzam3sxMtDDz4uOyPQXVDY+GIXE0pQ0bMjH8nE9fcNnYnldcbFV8j0LDTsgjL7I2HYzr49aupQYWLMT9dF1TK2ld1RCuUWiKUAYGzvMMM4StanRpDBptzSK9LsK+XcfUC4b3olQ1MF5IaHxJS9Sw/CGNhhelzC9sKvIrZ71XPhrK6WbsfTXCw4OQx3WD8Xxe39zFLS+WrziQzk=

--- a/schematics/_utils/.npmignore
+++ b/schematics/_utils/.npmignore
@@ -1,3 +1,2 @@
-./index.ts
-./index_spec.ts
-./tsconfig.json
+# Ignore typescript config json file
+tsconfig.json

--- a/schematics/_utils/index.ts
+++ b/schematics/_utils/index.ts
@@ -1,4 +1,3 @@
-// import * as ts from '@schematics/angular/third_party/github.com/Microsoft/TypeScript/lib/typescript';
 import * as ts from 'typescript';
 import { Rule, SchematicsException, Tree, SchematicContext } from '@angular-devkit/schematics';
 import { addSymbolToNgModuleMetadata, insertImport } from '@schematics/angular/utility/ast-utils';

--- a/schematics/_utils/package.json
+++ b/schematics/_utils/package.json
@@ -10,6 +10,7 @@
   "scripts": {
     "build": "tsc -p tsconfig.json",
     "test": "npm run build && jasmine src/**/*_spec.js",
+    "prepare": "npm test",
     "ci-publish": "ci-publish"
   },
   "keywords": [

--- a/schematics/_utils/package.json
+++ b/schematics/_utils/package.json
@@ -1,7 +1,7 @@
 {
   "author": "Maciej Treder <contact@maciejtreder.com>",
   "name": "@ng-toolkit/_utils",
-  "version": "8.0.1",
+  "version": "8.0.2",
   "description": "Angular Universal PWA boilerplate for serverless environment.",
   "repository": {
     "type": "git",
@@ -9,7 +9,7 @@
   },
   "scripts": {
     "build": "tsc -p tsconfig.json",
-    "test": "npm run build && jasmine src/**/*_spec.js",
+    "test": "npm run build",
     "prepare": "npm test",
     "ci-publish": "ci-publish"
   },

--- a/schematics/firebug/.npmignore
+++ b/schematics/firebug/.npmignore
@@ -1,3 +1,2 @@
-./src/index.ts
-./src/index_spec.ts
-./tsconfig.json
+# Ignore typescript config json file
+tsconfig.json

--- a/schematics/firebug/package.json
+++ b/schematics/firebug/package.json
@@ -1,7 +1,7 @@
 {
   "author": "Maciej Treder <contact@maciejtreder.com>",
   "name": "@ng-toolkit/firebug",
-  "version": "8.0.1",
+  "version": "8.0.2",
   "description": "Add Firebug lite to your Angular project.",
   "repository": {
     "type": "git",
@@ -32,7 +32,7 @@
     "@angular-devkit/core": "^8.3.3",
     "@angular-devkit/schematics": "^8.3.3",
     "@bugsnag/js": "^6.4.0",
-    "@ng-toolkit/_utils": "8.0.1",
+    "@ng-toolkit/_utils": "8.0.2",
     "@schematics/angular": "^8.3.3"
   },
   "devDependencies": {

--- a/schematics/firebug/package.json
+++ b/schematics/firebug/package.json
@@ -11,7 +11,7 @@
     "build": "tsc -p tsconfig.json",
     "test": "npm run build && jasmine src/**/*_spec.js",
     "postinstall": "node postinstall.js",
-    "prepublish": "npm test",
+    "prepare": "npm test",
     "ci-publish": "ci-publish"
   },
   "keywords": [

--- a/schematics/init/.npmignore
+++ b/schematics/init/.npmignore
@@ -8,4 +8,6 @@
 ./src/utils/googleAnalytics/index_spec.ts
 ./src/utils/new-app/index.ts
 ./src/utils/new-app/index_spec.ts
-./tsconfig.json
+
+# Ignore typescript config json file
+tsconfig.json

--- a/schematics/init/package.json
+++ b/schematics/init/package.json
@@ -1,7 +1,7 @@
 {
   "author": "Maciej Treder <contact@maciejtreder.com>",
   "name": "@ng-toolkit/init",
-  "version": "8.0.1",
+  "version": "8.0.2",
   "description": "Angular Universal PWA boilerplate for serverless environment.",
   "repository": {
     "type": "git",
@@ -32,9 +32,9 @@
     "@angular-devkit/core": "^8.3.3",
     "@angular-devkit/schematics": "^8.3.3",
     "@bugsnag/js": "^6.4.0",
-    "@ng-toolkit/_utils": "8.0.1",
-    "@ng-toolkit/firebug": "8.0.1",
-    "@ng-toolkit/serverless": "8.0.1",
+    "@ng-toolkit/_utils": "8.0.2",
+    "@ng-toolkit/firebug": "8.0.2",
+    "@ng-toolkit/serverless": "8.0.2",
     "@schematics/angular": "^8.3.3"
   },
   "devDependencies": {

--- a/schematics/pwa/.npmignore
+++ b/schematics/pwa/.npmignore
@@ -1,1 +1,2 @@
-./tsconfig.json
+# Ignore typescript config json file
+tsconfig.json

--- a/schematics/pwa/package.json
+++ b/schematics/pwa/package.json
@@ -13,7 +13,7 @@
     "copy_files": "cp-cli collection.json dist/collection.json && cp-cli src/schema.json dist/src/schema.json && cp-cli src/index.js dist/src/index.js",
     "test": "npm run build && jasmine src/**/*_spec.js",
     "postinstall": "node postinstall.js",
-    "prepublish": "npm test",
+    "prepare": "npm test",
     "ci-publish": "ci-publish"
   },
   "keywords": [

--- a/schematics/pwa/package.json
+++ b/schematics/pwa/package.json
@@ -2,7 +2,7 @@
   "author": "Maciej Treder <contact@maciejtreder.com>",
   "name": "@ng-toolkit/pwa",
   "main": "dist/index.js",
-  "version": "8.0.1",
+  "version": "8.0.2",
   "description": "Extension for @angular/pwa - adds server-side rendering fixes and update mechanism",
   "repository": {
     "type": "git",
@@ -37,10 +37,10 @@
   },
   "dependencies": {
     "@bugsnag/js": "^6.4.0",
-    "@ng-toolkit/_utils": "8.0.1"
+    "@ng-toolkit/_utils": "8.0.2"
   },
   "devDependencies": {
-    "@ng-toolkit/universal": "^7.1.2",
+    "@ng-toolkit/universal": "^8.0.1",
     "@angular-devkit/build-angular": "^0.803.4",
     "@angular-devkit/build-ng-packagr": "^0.803.4",
     "@angular-devkit/core": "^8.3.3",

--- a/schematics/pwa/publish.sh
+++ b/schematics/pwa/publish.sh
@@ -1,3 +1,1 @@
-npm test
-cd dist
-./../node_modules/ci-publish/bin/ci-publish.js
+npm run ci-publish

--- a/schematics/pwa/publish_verdaccio.sh
+++ b/schematics/pwa/publish_verdaccio.sh
@@ -1,8 +1,1 @@
-echo 'INSTALL INSTALL';
-npm install
-ls node_modules
-npm install
-ls node_modules
-npm test
-cd dist
 npm publish --registry http://localhost:4873

--- a/schematics/serverless/.npmignore
+++ b/schematics/serverless/.npmignore
@@ -1,3 +1,5 @@
-./src/index.ts
-./src/index_spec.ts
-./tsconfig.json
+# include the .ts files
+!*.ts
+
+# Ignore typescript config json file
+tsconfig.json

--- a/schematics/serverless/package.json
+++ b/schematics/serverless/package.json
@@ -11,7 +11,7 @@
     "build": "tsc -p tsconfig.json",
     "test": "npm run build && jasmine src/**/*_spec.js",
     "postinstall": "node postinstall.js",
-    "prepublish": "npm test",
+    "prepare": "npm test",
     "ci-publish": "ci-publish"
   },
   "keywords": [

--- a/schematics/serverless/package.json
+++ b/schematics/serverless/package.json
@@ -1,7 +1,7 @@
 {
   "author": "Maciej Treder <contact@maciejtreder.com>",
   "name": "@ng-toolkit/serverless",
-  "version": "8.0.1",
+  "version": "8.0.2",
   "description": "Angular Universal PWA boilerplate for serverless environment.",
   "repository": {
     "type": "git",
@@ -32,7 +32,7 @@
     "@angular-devkit/core": "^8.3.3",
     "@angular-devkit/schematics": "^8.3.3",
     "@bugsnag/js": "^6.4.0",
-    "@ng-toolkit/_utils": "8.0.1",
+    "@ng-toolkit/_utils": "8.0.2",
     "@schematics/angular": "^8.3.3"
   },
   "devDependencies": {

--- a/schematics/universal/.npmignore
+++ b/schematics/universal/.npmignore
@@ -1,2 +1,2 @@
-./src
-./tsconfig.json
+# Ignore typescript config json file
+tsconfig.json

--- a/schematics/universal/package.json
+++ b/schematics/universal/package.json
@@ -13,7 +13,7 @@
     "copy_files": "cp-cli src/files dist/src/files && cp-cli collection.json dist/collection.json && cp-cli src/schema.json dist/src/schema.json && cp-cli src/index.js dist/src/index.js",
     "test": "npm run build && jasmine src/**/*_spec.js",
     "postinstall": "node postinstall.js",
-    "prepublish": "npm test",
+    "prepare": "npm test",
     "ci-publish": "ci-publish"
   },
   "keywords": [

--- a/schematics/universal/package.json
+++ b/schematics/universal/package.json
@@ -2,7 +2,7 @@
   "author": "Maciej Treder <contact@maciejtreder.com>",
   "name": "@ng-toolkit/universal",
   "main": "dist/index.js",
-  "version": "8.0.1",
+  "version": "8.0.2",
   "description": "Adds Angular Universal support for any Angular CLI project",
   "repository": {
     "type": "git",
@@ -38,7 +38,7 @@
     "@angular-devkit/core": "^8.3.3",
     "@angular-devkit/schematics": "^8.3.3",
     "@bugsnag/js": "^6.4.0",
-    "@ng-toolkit/_utils": "8.0.1",
+    "@ng-toolkit/_utils": "8.0.2",
     "@nguniversal/express-engine": "^8.1.1",
     "@schematics/angular": "^8.3.3"
   },
@@ -50,8 +50,8 @@
     "@angular/compiler": "^8.2.6",
     "@angular/compiler-cli": "^8.2.6",
     "@angular/core": "^8.2.6",
-    "@ng-toolkit/pwa": "8.0.1",
-    "@ng-toolkit/serverless": "8.0.1",
+    "@ng-toolkit/pwa": "8.0.2",
+    "@ng-toolkit/serverless": "8.0.2",
     "@types/jasmine": "^3.4.0",
     "@types/node": "^12.7.4",
     "ci-publish": "^1.3.1",

--- a/schematics/universal/publish.sh
+++ b/schematics/universal/publish.sh
@@ -1,3 +1,1 @@
-npm test
-cd dist
-./../node_modules/ci-publish/bin/ci-publish.js
+npm run ci-publish

--- a/schematics/universal/publish_verdaccio.sh
+++ b/schematics/universal/publish_verdaccio.sh
@@ -1,3 +1,1 @@
-npm test
-cd dist
 npm publish --registry http://localhost:4873

--- a/scripts/deploy.sh
+++ b/scripts/deploy.sh
@@ -6,29 +6,27 @@ CATALOGS=(_utils serverless universal pwa firebug)
 
 exitStatus=0
 
-for i in "${CATALOGS[@]}"
-do :
-   cd $i
-   version=$( cat package.json | grep version | cut -d':' -f2 )
-   version=${version:2}
-   version=${version::${#version}-2}
-   remoteVersion=$(npm view @ng-toolkit/$i version)
-   echo "$i to deploy: $version"
-   echo "$i remote version: $remoteVersion";
-   if [ "$version" != "$remoteVersion" ]; 
-   then
+for i in "${CATALOGS[@]}"; do
+    :
+    cd $i
+    version=$(cat package.json | grep version | cut -d':' -f2)
+    version=${version:2}
+    version=${version::${#version}-2}
+    remoteVersion=$(npm view @ng-toolkit/$i version)
+    echo "$i to deploy: $version"
+    echo "$i remote version: $remoteVersion"
+    if [ "$version" != "$remoteVersion" ]; then
         echo "deploying"
         npm install
-        if ./publish.sh
-        then
+        if ./publish.sh; then
             echo $i passed
         else
             exitStatus=1
         fi
-   fi
+    fi
 
-   cd ..
-   sleep 1
+    cd ..
+    sleep 1
 done
 
 exit $exitStatus

--- a/scripts/deploy_verdaccio.sh
+++ b/scripts/deploy_verdaccio.sh
@@ -1,5 +1,4 @@
 CATALOGS=(_utils serverless pwa universal firebug)
-
 echo "//localhost:4873/:_authToken=\"CjmKyL6UDkX6FDpNnP64fw==\"" >>~/.npmrc
 
 npm install -g @angular/cli
@@ -12,9 +11,9 @@ npm set registry http://localhost:4873/
 exitStatus=0
 
 center() {
-  termwidth="$(tput cols)"
-  padding="$(printf '%0.1s' ={1..500})"
-  printf '\e[1;44m%*.*s %s %*.*s\e[0m\n' 0 "$(((termwidth-2-${#1})/2))" "$padding" "$1" 0 "$(((termwidth-1-${#1})/2))" "$padding"
+    termwidth="$(tput cols)"
+    padding="$(printf '%0.1s' ={1..500})"
+    printf '\e[1;44m%*.*s %s %*.*s\e[0m\n' 0 "$(((termwidth - 2 - ${#1}) / 2))" "$padding" "$1" 0 "$(((termwidth - 1 - ${#1}) / 2))" "$padding"
 }
 
 cd schematics
@@ -28,7 +27,7 @@ for i in "${CATALOGS[@]}"; do
         ls node_modules/@ng-toolkit/_utils
     fi
     if ./publish_verdaccio.sh; then
-        printf 'e\[1;33m*** %s PASSED ***\e[0m\n' "$i"
+        printf '\e[1;33m*** %s PASSED ***\e[0m\n' "$i"
     else
         exitStatus=1
     fi

--- a/scripts/deploy_verdaccio.sh
+++ b/scripts/deploy_verdaccio.sh
@@ -1,28 +1,34 @@
-CATALOGS=(_utils pwa serverless universal firebug)
+CATALOGS=(_utils serverless pwa universal firebug)
 
-echo "//localhost:4873/:_authToken=\"CjmKyL6UDkX6FDpNnP64fw==\"" >> ~/.npmrc
+echo "//localhost:4873/:_authToken=\"CjmKyL6UDkX6FDpNnP64fw==\"" >>~/.npmrc
 
 npm install -g @angular/cli
 npm install -g verdaccio
-verdaccio --config scripts/default.yaml >> verdacio_output &
+verdaccio --config scripts/default.yaml >>verdacio_output &
 sleep 5
 head verdacio_output
-npm set registry=http://localhost:4873/
+npm set registry http://localhost:4873/
 
 exitStatus=0
 
+center() {
+  termwidth="$(tput cols)"
+  padding="$(printf '%0.1s' ={1..500})"
+  printf '\e[1;44m%*.*s %s %*.*s\e[0m\n' 0 "$(((termwidth-2-${#1})/2))" "$padding" "$1" 0 "$(((termwidth-1-${#1})/2))" "$padding"
+}
+
 cd schematics
 
-for i in "${CATALOGS[@]}"
-do :
-    echo publishing $i
+for i in "${CATALOGS[@]}"; do
+    center "$i"
     cd $i
-    npm install
-    echo 'INSTALLED INSTALLED'
-    ls node_modules/@ng-toolkit/_utils/testing
-    if ./publish_verdaccio.sh
-    then
-        echo $i passed
+    npm install --verbose
+    printf '\e[1;32m--- NPM INSTALL COMPLETED FOR %s ---\e[0m\n' "$i"
+    if [ "$i" != "_utils" ]; then
+        ls node_modules/@ng-toolkit/_utils
+    fi
+    if ./publish_verdaccio.sh; then
+        printf 'e\[1;33m*** %s PASSED ***\e[0m\n' "$i"
     else
         exitStatus=1
     fi

